libdnf/utils/locker.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "locker.hpp"
+
+#include "libdnf/common/exception.hpp"
+
+#include <errno.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+namespace libdnf::utils {
+
+Locker::Locker(const std::string & name) {
+    // unix domain socket path is declared as char[108]
+    libdnf_assert(name.size() <= 107, "Lock name is too large (maximum length 107 expected)");
```

Comment:
Using this magic constant seems dangerous.
Yes `108` on Linux. But only `104` on FreeBSD.
How about using real value instead of a magic number?
```
constexpr int max_path_len = (sizeof sockaddr_un::sun_path) - 1;
libdnf_assert(name.size() <= max_path_len, "Lock name is too large (maximum length {} expected)", max_path_len);
```

Reply:
I don't think this is a case for an assert either, is it? The path can ultimately be influenced by user input, like in a configuration, maybe it's inside an installroot, right? In that case we don't want a crash, this situation can occur normally.

Reply:
I'm not sure here about the assert. It's not supposed to be a public API so users should not call this directly. The `path` in this case is just arbitrary identifier of the lock, it's not a file path directly influenced by installroot. Before calling a lock we need to construct this identifier and make sure it's length is shorter than the maximum.